### ISSUE TITLE
_price not update on date the _sales_price should be active

### ISSUE THREAD

**Issue by yulumadavid** [woocommerce/woocommerce#25222]
**Describe the bug**
When setting a sales price for tomorrow this price is added in the interface. But the next day (when the sales price should become active) the normale price has a strikethroug BUT the price remains the same. When looking at the database the _regular_price is set to the correct price, the _sale_price is set to the correct price BUT the _price value is NOT set to the right value.

On the day we create the salesprice it works fine, but the moment the clock's passes 00:00:00 the price won't update.

This counts aswell for salesprices set for tomorrow as for salesprices that end today. After 6 hours of debuging my conclusion is that the _price field is not updated. I have no clue at all what could trigger this.

I would love to test with plugins and theme's turned of but it would be so much nicer if I just know what process would trigger the update of the _price field.

Important: i've spend also a lot of time with our clean install, and can't reproduce this problem. We run over aprox 100 Woo stores and only experience this problem with 4 webshops. We can't see any similarities for only these 4 stores.

If any one could point us what process is updating the _price field we could test and debug in the right way so that we could test WHAT plugin, code or Theme creates this problem.

**Comment by yulumadavid** [comment]
PS: i've thought it could be this daily cron? 'woocommerce_scheduled_sales' but would like to know a bit more to make debug/loggin possible.

**Comment by juliaamosova** [comment]
@yulumadavid you are correct that the action `woocommerce_scheduled_sales` is what triggers the price changes when a scheduled sale starts. Besides that, I don't think there is much we can do to help without further troubleshooting on your end. 

It is tricky indeed that the issue happens only 4 stores for you. Perhaps, you can make copies/staging sites of these 4 sites and observe there further with all plugins deactivated and theme switched to default Storefront.

**Comment by tammullen** [comment]
> Important: i've spend also a lot of time with our clean install, and can't reproduce this problem. We run over aprox 100 Woo stores and only experience this problem with 4 webshops. We can't see any similarities for only these 4 stores.

I also was unable to reproduce this using a clean install.

The sale price updated both normally and on schedule including the scenario of setting for next day and it updating as expected at 00:00 of that day.

<img width="373" alt="Screenshot 2019-12-16 at 10 55 20" src="https://user-images.githubusercontent.com/24649833/70901613-08994d80-1ff3-11ea-9d46-123b7b74c656.png">
<img width="284" alt="Screenshot 2019-12-16 at 10 54 56" src="https://user-images.githubusercontent.com/24649833/70901615-09ca7a80-1ff3-11ea-9a95-027a09a18188.png">

**Comment by yulumadavid** [comment]
Testing right now. Can't reproduce when triggering the cron manual so have to test for tomorrow and so on. Keep you posted or will update this issue when I have a solution.

**Comment by juliaamosova** [comment]
Hi @yulumadavid,

As a part of this repository’s maintenance, I am closing this issue due to inactivity. Please feel free to comment on it in case we missed something. We’d be happy to take another look.

**Comment by yulumadavid** [comment]
Sometimes WP works in mysterious ways. Promised to keep you updates.

A few days later we tested it again on our test webshop and the problem did no longer exists. We then tested it on the four webshops where the problem occurred, and there also the problem did not occur anymore. I can only assume that the daily cron was for some reason blocked for only these shops for only a few days of a period of time.

case closed, thanks everyone for responding!
